example/lib/main.dart

```diff
@@ -280,90 +281,76 @@ class MyHomePageState extends State<MyHomePage> {
-            mathMatcher():
-                mathRender(onMathError: (error, exception, exceptionWithType) {
-              debugPrint(exception);
-              return Text(exception);
-            }),
-            svgTagMatcher(): svgTagRender(),
-            svgDataUriMatcher(): svgDataImageRender(),
-            svgAssetUriMatcher(): svgAssetImageRender(),
-            svgNetworkSourceMatcher(): svgNetworkImageRender(),
-            networkSourceMatcher(domains: ["flutter.dev"]):
-                CustomRender.widget(widget: (context, buildChildren) {
-              return const FlutterLogo(size: 36);
-            }),
-            networkSourceMatcher(domains: ["mydomain.com"]): networkImageRender(
-              headers: {"Custom-Header": "some-value"},
-              altWidget: (alt) => Text(alt ?? ""),
-              loadingWidget: () => const Text("Loading..."),
+            'flutter': Style(
+              display: Display.block,
+              fontSize: FontSize(5, Unit.em),
             ),
-            // On relative paths starting with /wiki, prefix with a base url
-            (context) =>
-                context.tree.element?.attributes["src"] != null &&
-                context.tree.element!.attributes["src"]!
-                    .startsWith("/wiki"): networkImageRender(
-                mapUrl: (url) => "https://upload.wikimedia.org${url!}"),
-            // Custom placeholder image for broken links
-            networkSourceMatcher():
-                networkImageRender(altWidget: (_) => const FlutterLogo()),
-            videoMatcher(): videoRender(),
           },
-          onLinkTap: (url, _, __, ___) {
+          extensions: [
+            TagExtension(
+              tagsToExtend: {"tex"},
+              builder: (context) => Math.tex(
+                context.innerHtml,
+                mathStyle: MathStyle.display,
+                textStyle: context.styledElement?.style.generateTextStyle(),
+                onErrorFallback: (FlutterMathException e) {
+                  return Text(e.message);
+                },
+              ),
+            ),
+            TagExtension.inline(
+              tagsToExtend: {"bird"},
+              child: const TextSpan(text: "🐦"),
+            ),
+            TagExtension(
+              tagsToExtend: {"flutter"},
+              builder: (context) => FlutterLogo(
+                style: context.attributes['horizontal'] != null
+                    ? FlutterLogoStyle.horizontal
+                    : FlutterLogoStyle.markOnly,
+                textColor: context.styledElement!.style.color!,
+                size: context.styledElement!.style.fontSize!.value,
+              ),
+            ),
+            const MathHtmlExtension(),
+            const AudioHtmlExtension(),
+            const VideoHtmlExtension(),
+            const IframeHtmlExtension(),
+            const TableHtmlExtension(),
+            const SvgHtmlExtension(),
+          ],
+          // customRenders: {
```

Comment:
This seems like it can be removed from the example.

Reply:
Good catch! I moved some of the customRenders to Extensions to show how that would be done, and removed the rest.